plugins/action_increment_stage/src/incrementStage.ts

```diff
@@ -1,29 +1,26 @@
-import { ActionPluginOutput } from '../../types'
+import { ActionPluginOutput, ActionPluginInput } from '../../types'
 
-type IParameters = {
-  applicationId: number
-}
-
-async function incrementStage(
-  parameters: IParameters,
-  DBConnect: any
-): Promise<ActionPluginOutput> {
-  const { applicationId } = parameters
+async function incrementStage({
+  parameters,
+  applicationData,
+  DBConnect,
+}: ActionPluginInput): Promise<ActionPluginOutput> {
+  const applicationId = parameters?.applicationId ?? applicationData?.applicationId
   const returnObject: ActionPluginOutput = { status: null, error_log: '' }
   console.log(`Incrementing the Stage for Application ${applicationId}...`)
 
   try {
-    const templateId: number = await DBConnect.getTemplateIdFromTrigger(
-      'application',
-      applicationId
-    )
-
-    console.log('Getting template Id', templateId)
+    const templateId: number =
```

Comment:
Maybe should be also listed as part of `optional parameters`?

Reply:
```
const templateId: number =
      applicationData?.templateId ??
      (await DBConnect.getTemplateIdFromTrigger('application', applicationId))
```
This is actually a bit of over-kill, as templateId will ALWAYS be in applicationData (as it's the first thing it needs to look up to get everything else). This is just a back-up query that's left over from earlier when we didn't have applicationData, but I thought it didn't hurt to leave it in. Definitely not going to need it in parameters, though ;)

Reply:
Yep. Agreed